src/Symfony/Component/HttpFoundation/ProblemDetailsJsonResponse.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+use Symfony\Component\HttpFoundation\Exception\ProblemDetailsJsonResponseException;
+
+/**
+ * Represents a JSON response with a Problem Details object.
+ *
+ * @author Abdellah Ramadan <ramadanabdel24@gmail.com>
+ */
+class ProblemDetailsJsonResponse extends Response
+{
+    public function __construct(
+        private ?int $status = null,
+        private ?string $title = null,
+        private readonly ?string $type = null,
+        private readonly ?string $detail = null,
+        private readonly ?string $instance = null,
+        private readonly ?array $extensions = [],
```

Comment:
why do we need to have all these properties if we never access them after the constructor?

Reply:
If they're not used after construct then they can be regular parameters

Reply:
Thank you @chalasr